src/scripts/main.js

```diff
@@ -1,3 +1,26 @@
 'use strict';
 
 // write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('table');
+  const headers = table.querySelectorAll('th');
+  const tableBody = table.querySelector('tbody');
+  const rows = Array.from(tableBody.querySelectorAll('tr'));
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      const sortedRows = rows.sort((a, b) => {
+        const cellA = a.querySelectorAll('td')[index].textContent.trim();
+        const cellB = b.querySelectorAll('td')[index].textContent.trim();
+
+        return cellA.localeCompare(cellB, undefined, { numeric: true });
+      });
```

Comment:
The sorting logic here directly modifies the `rows` array, which is a reference to the original rows in the table. This can lead to unexpected behavior if the sorting is performed multiple times, as the `rows` array will already be sorted from previous operations. Consider creating a new array for `sortedRows` using `Array.from(rows)` before sorting to avoid modifying the original `rows` array.